jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py

```diff
@@ -305,15 +303,17 @@ def calculate_moment(self, add_data=True):
             ref_wavelength = self.reference_wavelength * u.Unit(self.dataset_spectral_unit)
             slab_sa = slab.spectral_axis.to("km/s", doppler_convention="relativistic",
                                             doppler_rest=ref_wavelength)
-            slab = Spectrum1D(slab.flux, slab_sa, uncertainty=slab.uncertainty)
+            slab = Spectrum(slab.flux, slab_sa, uncertainty=slab.uncertainty,
+                            spectral_axis_index=cube.spectral_axis_index)
         # Otherwise convert spectral axis to display units, have to do frequency <-> wavelength
         # before calculating
         else:
             slab_sa = slab.spectral_axis.to(self.app._get_display_unit('spectral'))
-            slab = Spectrum1D(slab.flux, slab_sa, uncertainty=slab.uncertainty)
+            slab = Spectrum(slab.flux, slab_sa, uncertainty=slab.uncertainty,
+                            spectral_axis_index=cube.spectral_axis_index)
 
         # Finally actually calculate the moment
-        self.moment = analysis.moment(slab, order=n_moment).T
+        self.moment = analysis.moment(slab, order=n_moment)
```

Comment:
is this because `Spectrum` is being passed in transposed already so the output doesn't need to be?

Reply:
I think it's the opposite, it's because we're _not_ transposing the spectrum anymore. A second visual sanity check (I know I did when I was coding this) to make sure the output moment lines up would be prudent though if you didn't do so.